rust/cargo-psibase/src/config.rs

```diff
@@ -0,0 +1,70 @@
+use anyhow::{anyhow, Error};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use url::Url;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct PsibaseConfig {
+    pub hosts: HashMap<String, String>,
+}
+
+fn psibase_config_path() -> Result<PathBuf, Error> {
+    let home = std::env::var("HOME")?;
+    let path = Path::new(&home).join(".psibase.toml");
+    Ok(path)
+}
+
+fn read_psibase_config() -> Option<PsibaseConfig> {
+    // TODO: watch https://github.com/rust-lang/libs-team/issues/372
+    psibase_config_path()
+        .ok()
+        .map(|config_path| std::fs::read_to_string(&config_path).ok())
+        .flatten()
+        .map(|config_str| toml::from_str::<PsibaseConfig>(&config_str).ok())
+        .flatten()
+}
+
+fn write_psibase_config(config: PsibaseConfig) -> Result<(), Error> {
+    let path = psibase_config_path()?;
+    let config_str = toml::to_string(&config)?;
+    std::fs::write(path, config_str)?;
+    Ok(())
+}
+
+pub fn read_host_config_url(host_key: &str) -> Result<String, Error> {
+    let psibase_config = read_psibase_config().expect(
```

Comment:
Completely replacing any errors messages with a fixed messages seems like a bad idea, given that [the main goal of TOML is to generate errors](https://github.com/madmurphy/libconfini/wiki/An-INI-critique-of-TOML#17-against-postels-law-by-design)